src/main/scala/com/microsoft/hyperspace/actions/CreateAction.scala

```diff
@@ -65,9 +65,20 @@ class CreateAction(
   }
 
   private def isValidIndexSchema(config: IndexConfig, schema: StructType): Boolean = {
+    // First we flatten the schema. Instead of having struct of leaves
+    // the flatten method will return a list of field names.
+    // The second step is escaping the field names as Parquet does not works well
+    // with field names that contains the `.` (dot).
+    // Given `struct(nested, struct(nst, struct(field1)))`, the fields variable
+    // will contain `Seq("nested__nst__field1")`.
+    val fields = SchemaUtils.escapeFieldNames(SchemaUtils.flatten(schema))
```

Comment:
If the original column has ".", will it be also escaped?

For example, `val df = Seq(1).toDF("a.b")`

Reply:
Given this nested class: 

```
case class Nested(nest: String, v: Int)
```

And this dataset:

```
val df = Seq(("a", "nxn1", Nested("n1", 1))).toDF("name", "nested.nest", "nested")
```

The schema of the dataset is this:

```
root
 |-- name: string (nullable = true)
 |-- nested.nest: string (nullable = true)
 |-- nested: struct (nullable = true)
 |    |-- nest: string (nullable = true)
 |    |-- v: integer (nullable = false)
```

Doing `df.select("nested.nest").limit(1)` will return `n1`, the value of `nest` inside the `Nested` class.
Doing `df.select("`nested.nest`").limit(1)` will return `nxn1`, the value in the top level field called `nested.nest`.

Given the two things above I would say that if the user wants to access a field name that contains a dot and that field is not nested then it should be enclosed by back ticks (` `` `).

Reply:
I added the check for backticks and when flattening (`flatten` method) if we find a field name that is not nested but has `.` (dot) in the name it will be enclosed in backticks. This way we can differentiate between the two.

Reply:
What if there's no nested structure (nested -> (nested, v), and call df.select("nest.nest").limit(1)?

Reply:
@sezruby if you do that first, the `flatten` method will enclose it with backticks, then `escapeFieldName` will NOT escape it as we check that the field is enclosed in backticks.

Reply:
Hmm, since we have few questions around the "normalization", have you also considered just prefixing some constant (e.g., `_hyperspace_nested_col.nested.x.y`, where `_hyperspace_nested_col` will be just prepended for a nested column).

BTW, I see bug fixes related to nested fields in OSS Spark regularly, so this is not straightforward to get it right 😄 , so thanks @andrei-ionescu / @sezruby for the discussion.

Reply:
I like the prefix suggestion. I'll try implementing it and keeping those `.` (dots). I hope it works.